yjit/src/asm/mod.rs

```diff
@@ -69,25 +81,84 @@ pub struct CodeBlock {
 
 impl CodeBlock {
     /// Make a new CodeBlock
-    pub fn new(mem_block: VirtualMem, outlined: bool) -> Self {
-        Self {
-            mem_size: mem_block.virtual_region_size(),
+    pub fn new(mem_block: Rc<RefCell<VirtualMem>>, page_size: usize, outlined: bool) -> Self {
+        let mem_size = mem_block.borrow().virtual_region_size();
+        let mut cb = Self {
             mem_block,
+            mem_size,
             write_pos: 0,
+            page_size,
+            page_end_offset: 20, // TODO: make it non Arm-specific
```

Comment:
Maybe we should rename `page_end_offset` => `page_end_reserve`? 🤔

Reply:
:+1: https://github.com/ruby/ruby/pull/6460/commits/2d7b060fcdc614e4bfe531b4eeb6fb53301f333f